### PR TITLE
Update RTD security docs

### LINKED ISSUE #3637: Improve security contact webpage
We need to improve our documentation about a user who found a security issue could contact us to report this vulnerability in a confidential way.

This page should be clear regarding how to report the issue, how submit a patch (wihtout making it public) and what to do receive feedback / discuss about the solution.

There is a page already but it's too poor: https://readthedocs.org/security/

### PR DESCRIPTION
**Pull request by davidfischer**
Fixes #3637

### CODE REVIEW

File: readthedocs/urls.py [hunk 1]
```diff
@@ -34,6 +34,8 @@
     url(r'^$', HomepageView.as_view(), name='homepage'),
     url(r'^support/', SupportView.as_view(), name='support'),
     url(r'^security/', TemplateView.as_view(template_name='security.html')),
+    url(r'^.well-known/security.txt',
```
Comment (humitos) [inline readthedocs/urls.py]: Do we need this? Who uses it? How?

(I read the RFC at https://tools.ietf.org/html/rfc5785 but I don't understand the use case)

Comment (davidfischer) [reply readthedocs/urls.py]: I should have put more details. A `security.txt` file is not yet a standard but may be. It has been submitted to become an RFC. The goal of it is that it is a standard place where a security researcher can find the right place to disclose an issue.

https://securitytxt.org/

File: docs/security.rst [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+* We will add the issue to our :ref:`security issue archive <security-issue-archive>`.
+
+.. _pgp-key:
+
+PGP key
+~~~~~~~
+
+You may use this `PGP key`_ to securely communicate with us and to verify signed messages you receive from us.
+
+.. _PGP key: http://pgp.mit.edu:11371/pks/lookup?op=get&search=0x71337C3047A1B066
```
Comment (ericholscher) [inline docs/security.rst]: I believe this is my PGP key, and I'm not confident I still have access to it. Perhaps we should generate a new one via keybase or something, perhaps that we can share with the team?

File: docs/security.rst [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    luSpsAyzljd2WIX4OVtbOodsJeBYutS2AxBO
+    =yZOS
+    -----END PGP PUBLIC KEY BLOCK-----
+
+.. _security-issue-archive:
+
+Security issue archive
+~~~~~~~~~~~~~~~~~~~~~~
+
+It's only a matter of time...
```
Comment (ericholscher) [inline docs/security.rst]: Yep :/

File: docs/security.rst (hunk 2 above)
Comment (davidfischer) [reply docs/security.rst]: Oh, if you don't have access to it that is definitely a problem. I'll create a new one and share the key.

**Comment by davidfischer** [comment]
The `security@` email is now live

File: readthedocs/templates/security.txt [hunk 4]
```diff
@@ -0,0 +1 @@
+Policy: https://docs.readthedocs.io/en/latest/security.html
```
Comment (humitos) [inline readthedocs/templates/security.txt]: Reading the official page, https://securitytxt.org/ I found that there are more fields we can add here:

```
Contact:
Encryption: 
Acknowledgements: 
Policy: 
Signature: 
Hiring: 
```

Contact, Ecryptation and Signature are good candidates I think.

**Comment by davidfischer** [comment]
I have generated the new GPG key and I'll push an update here once it has been received by the key servers.

**Comment by davidfischer** [comment]
The key has been updated and this is ready to go.

**Comment by ericholscher** [comment]
🎆

**Pull request merged**